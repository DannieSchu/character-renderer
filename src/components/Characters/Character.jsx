import React from 'react';
import PropTypes from 'prop-types';

const Character = ({ name, url }) => {
  return (
    <section>
      <h3>{name}</h3>
      <img src={url} alt={name} />
    </section>
  );
};

Character.propTypes = {
  name: PropTypes.string.isRequired,
  url: PropTypes.string.isRequired
};

export default Character;
