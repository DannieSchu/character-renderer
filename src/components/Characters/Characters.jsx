import React from 'react';
import PropTypes from 'prop-types';
import { Link } from 'react-router-dom';
import Character from './Character.jsx';

const Characters = ({ characters }) => {
  const characterItems = characters.map(character => (
    <li key={character.name}>
      <Link to={`/${character.name}`}>
        <Character {...character} />
      </Link>
    </li>
  ));

  return (
    <ul>
      {characterItems}
    </ul>
  );
};

Characters.propTypes = {
  characters: PropTypes.arrayOf(PropTypes.shape({
    name: PropTypes.string.isRequired,
    url: PropTypes.string.isRequired
  })).isRequired
};

export default Characters;
