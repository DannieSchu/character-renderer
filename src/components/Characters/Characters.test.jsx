import React from 'react';
import { shallow } from 'enzyme';
import Characters from './Characters.jsx';

describe('Characters', () => {
  it('matches a snapshop', () => {
    const characters = [{
      name: 'Helga G. Pataki',
      url: 'https://vignette.wikia.nocookie.net/heyarnold/images/e/e5/Helga_Pataki.png/revision/latest/scale-to-width-down/266?cb=20190816021807'
    },
    {
      name: 'Shelley',
      url: 'https://vignette.wikia.nocookie.net/heyarnold/images/c/cb/Shelley.jpg/revision/latest/scale-to-width-down/310?cb=20120527025256'
    }];
    const wrapper = shallow(
      <Characters characters={characters} />
    );
    expect(wrapper).toMatchSnapshot();
  });
});
