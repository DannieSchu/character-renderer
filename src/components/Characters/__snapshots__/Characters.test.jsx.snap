// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Characters matches a snapshop 1`] = `
<ul>
  <li
    key="Helga G. Pataki"
  >
    <Link
      to="/Helga G. Pataki"
    >
      <Character
        name="Helga G. Pataki"
        url="https://vignette.wikia.nocookie.net/heyarnold/images/e/e5/Helga_Pataki.png/revision/latest/scale-to-width-down/266?cb=20190816021807"
      />
    </Link>
  </li>
  <li
    key="Shelley"
  >
    <Link
      to="/Shelley"
    >
      <Character
        name="Shelley"
        url="https://vignette.wikia.nocookie.net/heyarnold/images/c/cb/Shelley.jpg/revision/latest/scale-to-width-down/310?cb=20120527025256"
      />
    </Link>
  </li>
</ul>
`;
